src/app/util/basic-types.h

```diff
@@ -0,0 +1,65 @@
+/**
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ * @file
+ * @brief The include file for all the types for the data model that are not
+ *        dependent on an individual application configuration.
+ */
+
+#pragma once
+
+#include <stdint.h>
+
+/**
+ * @brief Type for referring to zigbee application profile id
+ *        TODO: This is probably not needed for CHIP and should be removed.
+ *        https://github.com/project-chip/connectedhomeip/issues/3444
+ */
+typedef uint16_t EmberAfProfileId;
+
+/**
+ * @brief Type for referring to CHIP attribute id.
+ */
+typedef uint16_t CHIPAttributeId;
+typedef CHIPAttributeId EmberAfAttributeId;
+#ifdef __cplusplus
```

Comment:
in C++, there are 2 typedefs?  `chip::CHIPAttributeId` and `CHIPAttributeId`?

Reply:
Yes, sort of purposefully for now.  We can `CHIPAttributeId` to headers that are included from both C and C++ for the moment.   Once we stop having C source files, we can mass-rename easily.